src/js/models/layer.js

```diff
@@ -484,32 +594,37 @@ define(function (require, exports, module) {
      */
     Layer.fromGroupDescriptor = function (documentID, layerID, layerName, isGroupEnd,
         isArtboard, boundsDescriptor) {
-        return new Layer({
-            id: layerID,
-            key: documentID + "." + layerID,
-            name: isGroupEnd ? "</Layer Group>" : layerName,
-            kind: isGroupEnd ? layerLib.layerKinds.GROUPEND : layerLib.layerKinds.GROUP,
-            visible: true,
-            locked: false,
-            expanded: true,
-            isBackground: false,
-            opacity: 100,
-            selected: true, // We'll set selected after moving layers
-            fill: null,
-            stroke: null,
-            dropShadows: Immutable.List(),
-            innerShadows: Immutable.List(),
-            effects: LayerEffect.EMPTY_EFFECTS,
-            mode: "passThrough",
-            proportionalScaling: false,
-            isArtboard: !!isArtboard,
-            bounds: isArtboard ? new Bounds(boundsDescriptor) : null,
-            isLinked: false,
-            vectorMaskEnabled: false,
-            vectorMaskEmpty: true,
-            textWarningLevel: 0,
-            initialized: true
-        });
+        var Bounds = require("./bounds"),
```

Comment:
Sorry, I forgot to leave a comment on this. This is for resolving the circular dependencies between Layer and Bounds (also Fill). If it is put on top, `Bounds` will be an empty object as it is also requiring `Layer`. This is one of the recommended ways of dealing the situation (http://requirejs.org/docs/api.html#circular). 

Another way that will allow us putting the require statement on top, is exposing the model via the `export` object, just like what we have for actions. For example:

```
// bounds.js
var Layer = require("layer");
...
exports.Model = Bounds.

// layer.js
var Bounds = require("bounds");  // Bounds refers to the exports object, so change in the bounds.js also reflects in here.

function createLayer() {
   ...
   var bounds = new Bounds.Model();
}
```

NOTE: ES6 package has the `export default ...` feature that will solve this nice, but until we are ready for it, we may have keep using the require inside function.